language/ruby/language_test.go

```diff
@@ -62,3 +64,318 @@ func TestLanguageImportPath(t *testing.T) {
 		ExpectedImportPath: "../lib/plain",
 	})
 }
+
+func TestMistakes(t *testing.T) {
+	validate := func(t *testing.T, tc *languagetesting.TestCaseMistakes) {
+		tc.Validate(t)
+	}
+
+	validate(t, &languagetesting.TestCaseMistakes{
+		Name: "Argument is missing",
+
+		Language:       &Language{},
+		RepositoryPath: filepath.Join("..", "..", "testdata", "ruby", "mistakes", "argumentMissing"),
+
+		ExpectedMistakesContains: func(mistakes []string) {
+			assert.Contains(t, mistakes[0], "ArgumentError: wrong number of arguments (given 1, expected 0)")
+			assert.Contains(t, mistakes[0], "argument_missing.rb:1:in `argument_missing'")
+		},
+	})
+}
+
+func TestExtractMistakes(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		RawMistakes string
+
+		ExpectedMistakes func(mistakes []string)
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualMistakes := extractMistakes(bytesutil.StringTrimIndentations(tc.RawMistakes))
+
+			tc.ExpectedMistakes(actualMistakes)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Argument missing",
+
+		RawMistakes: `
+			Run options: --seed 51400
+
+			# Running:
+
+			EEE
+
+			Finished in 0.002350s, 1276.5192 runs/s, 0.0000 assertions/s.
+
+			1) Error:
+			TestArgumentMissing#test_argument_missing2:
+			ArgumentError: wrong number of arguments (given 1, expected 0)
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/lib/argument_missing.rb:1:in ` + "`" + `argument_missing'
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/test/argument_missing_test.rb:15:in ` + "`" + `test_argument_missing2'
+
+			2) Error:
+			TestArgumentMissing#test_argument_missing1:
+			ArgumentError: wrong number of arguments (given 1, expected 0)
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/lib/argument_missing.rb:1:in ` + "`" + `argument_missing'
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/test/argument_missing_test.rb:8:in ` + "`" + `test_argument_missing1'
+
+			3) Error:
+			TestArgumentMissing#test_argument_missing3:
+			ArgumentError: wrong number of arguments (given 1, expected 0)
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/lib/argument_missing.rb:1:in ` + "`" + `argument_missing'
+				/home/rui/eval-dev-quality/testdata/ruby/mistakes/argumentMissing/test/argument_missing_test.rb:22:in ` + "`" + `test_argument_missing3'
+
+			3 runs, 0 assertions, 0 failures, 3 errors, 0 skips
+			rake aborted!
+			Command failed with status (1)
+
+			Tasks: TOP => test
+			(See full trace by running task with --trace)
+		`,
+
+		ExpectedMistakes: func(mistakes []string) {
+			assert.Contains(t, mistakes[0], "ArgumentError: wrong number of arguments (given 1, expected 0)")
+			assert.Contains(t, mistakes[0], "argument_missing.rb:1:in `argument_missing'")
```

Comment:
Since this is the core of how we extract the error messages, I would like to see an actual 1:1 assertion, not just contains. Just to make sure that the extracted message does not contain too much junk.